docs/specifications/tests/Methods.md

```diff
@@ -1,74 +1,135 @@
-2. If the recursion reaches a name server that responds with a redirect
-   directly to the requested domain, including functional glue, the test
-   succeeds. The domain through which the name server was found is
-   considered the parent domain.  
-3. If the recursion reaches a name server that authoritatively responds
-   with NXDOMAIN for the child domain, the test succeeds. The domain through
-   which the name server was found is considered the parent domain.
+Obtain the parent zone of the given domain (zone).
+
+1. The given domain is assumed to be apex of a zone with the same
+   name, and is here refered to as the _child zone_.
+2. A recursive lookup for the SOA record of the child domain name starting from the
+   root zone is done, and the steps of the process are recorded.
+3. If the lookup reaches a name server that responds with a redirect (delegation)
+   directly to the requested child zone the method
+   succeeds. The zone in which the delegation was found is
+   considered to be the parent zone.  
+4. If the recursive lookup reaches a name server that authoritatively responds
+   with NXDOMAIN for the child domain (child zone), the method succeeds. The zone
+   returning NXDOMAIN is considered to be the parent zone.
+5. If the recursive lookup reaches authorititative NOERROR answer, with or with
+   record in the answer section, the method fails, and no parent zone is found.
+6. If the recurse lookup is exhausted or ends in a loop, then the method fails
+   and no parent zone is found.
 
 
 ## Method 2: Delegation name servers
 
-Obtain the authoritative name servers for the given zone as defined in
-its delegation in the parent zone.
+Obtain the name servers (NS records) for the given zone as defined in
+its delegation from the parent zone.
 
 1. Obtain parent zone using Method 1.
-2. Send an NS query for the given domain name to one of the authoritative
+
+A. Normal test. For undelegated test go to B.
+
+2. Send an SOA query for the given domain name (zone) to one of the authoritative
    name servers for the parent zone.
-3. Record all NS records in the authority section and all glue records.
+3. Record all A and AAAA records in the additional section and store in a cache that
+   method 4 can use.
+4. Record all NS records in the authority section and return the name server names
+   in the response.
+
+B. Undelegated test.
+
+2. If the test is an undelegated test, then the name servers names in the input
+   will be used as the equivalent of RDATA of the NS records from the lookup 
+   under A.
+3. Collect all IPv4 and IPv6 addresses in the input to the undelegated test and 
+   store in a cache where the connection between name server name and IP address
+   is kept. The cache is to be used by method 4.
 
 
 ## Method 3: In-zone name servers
 
-Obtain the authoritative name servers for the given zone as defined in
-the zone itself.
+Obtain the authoritative name servers (NS) for the given zone (child zone) as 
```

Comment:
Is the "(child zone)" addition needed for disambiguation?

Reply:
I think it makes the risk of confusion lower.